plugins/modules/inventory_intent.py

```diff
@@ -713,59 +985,232 @@ def get_diff_merged(self, config):
+                        temp_params = {
+                            'description': interface_params.get('description', ''),
+                            'adminStatus': interface_params.get('admin_status'),
+                            'voiceVlanId': interface_params.get('voice_vlan_id'),
+                            'vlanId': interface_params.get('vlan_id')
+                        }
+                        payload_params = {}
+                        for key, value in temp_params.items():
+                            if value is not None:
+                                payload_params[key] = value
+
+                        update_interface_params = {
+                            'payload': payload_params,
+                            'interface_uuid': interface_id,
+                            'deployment_mode': interface_params.get('deployment_mode', 'Deploy')
+                        }
+                        response = self.dnac._exec(
+                            family="devices",
+                            function='update_interface_details',
+                            op_modifies=True,
+                            params=update_interface_params,
+                        )
+                        log(str(response))
+
+                        if response and isinstance(response, dict):
+                            task_id = response.get('response').get('taskId')
+                            while True:
+                                execution_details = self.get_task_details(task_id)
+
+                                if 'SUCCESS' in execution_details.get("progress"):
+                                    self.status = "success"
+                                    self.result['changed'] = True
+                                    self.result['response'] = execution_details
+                                    log_msg = "Update Interface Details for device {0} Added Successfully".format(device_ip)
+                                    log(log_msg)
+                                    self.result['log'].append(log_msg)
+                                    break
+                                elif execution_details.get("isError") and execution_details.get("failureReason"):
+                                    self.msg = "Interface Updation get failed because of {0}".format(execution_details.get("failureReason"))
+                                    self.status = "failed"
+                                    break
+
+                    except Exception as e:
+                        error_message = "Error while Updating Interface Details in Cisco DNA Center - {0}".format(str(e))
+                        self.status = "success"
+                        self.result['changed'] = False
+                        log_msg = "Port actions are only supported on user facing/access ports as it's not allowed or No Updation required"
+                        self.result['log'] = log_msg
+
+                return self
+
+            else:
```

Comment:
as we are returning from if, do we need this else?

Reply:
chnaged.